datafusion/common/src/scalar/mod.rs

```diff
@@ -3399,6 +3451,43 @@ impl fmt::Display for ScalarValue {
                         .join(",")
                 )?
             }
+            ScalarValue::Map(map_arr) => {
+                if map_arr.null_count() == map_arr.len() {
```

Comment:
should this also have the assert like Struct? 
```
// ScalarValue Map should always have a single element
assert_eq!(map_arr.len(), 1);
```

Reply:
or if not, then maybe https://github.com/apache/datafusion/pull/11224/files#diff-49e275af8f09685c7bbc491db8ab3b9479960878f42ac558ec0e3e39570590bdR3583 shoulnd't have it either ? 😅

Reply:
> should this also have the assert like Struct?
> 
> ```
> // ScalarValue Map should always have a single element
> assert_eq!(map_arr.len(), 1);
> ```


No, `MapArray` is `StructArray`. It could contain more than one element. 

> or if not, then maybe https://github.com/apache/datafusion/pull/11224/files#diff-49e275af8f09685c7bbc491db8ab3b9479960878f42ac558ec0e3e39570590bdR3583 shoulnd't have it either ? 😅

I forgot to remove this. Thanks for reminding me.

Reply:
I add some tests to cover the Debug function in the latest commit.